### DISCUSSION TITLE
Create clickable H1 html tags

### DISCUSSION THREAD

**Discussion by abd3lraouf** [varabyte/kobweb#478] [Q&A]
What if I wanted to add a link to the generated H1 tag

so 
```
# Title goes here
```

can be converted to

```kotlin
H1 {
    Link("title-goes-here") {
        Text("Kobweb Tutorial")
    }
}
```

**Comment by bitspittle** [comment]
I do something like that in my blog site
https://github.com/bitspittle/bitspittle.dev/blob/51146a1216fa91615c0a792162111940a795484c/site/build.gradle.kts#L70

Note that it includes a widget I defined here: <https://github.com/bitspittle/bitspittle.dev/blob/main/site/src/jsMain/kotlin/dev/bitspittle/site/components/widgets/navigation/HoverLink.kt>

Of course, you can leave that out on a first pass

P.S. Note that the base markdown support already adds IDs to header tags (<https://github.com/varabyte/kobweb/blob/0a86781ea3eb977d0762badd4fd0f3785bcbeb2a/tools/gradle-plugins/extensions/markdown/src/main/kotlin/com/varabyte/kobwebx/gradle/markdown/MarkdownHandlers.kt#L212>). You can check out the Toc widget in Silk which takes advantage of this.

**Reply by abd3lraouf** [reply]
Thank you so much @bitspittle for the assistance I will check soon
